frontend/app/utilities/fetchapi.ts

```diff
@@ -1,219 +1,305 @@
+  );
+  if (!response.ok) {
+    if (response.status === 404) {
+      throw new Error(`Tag with name "${tagName}" not found.`);
     }
+    throw new Error("Failed to delete tag");
+  }
 };
 
 // Get all tags for a mission
 export const getTagsByMission = async (missionId: number): Promise<Tag[]> => {
-    const response = await fetch(`${FETCH_API_BASE_URL}/missions/tags/${missionId}`, {
-        method: 'GET',
-        headers: headers,
-    });
-    if (!response.ok) {
-        if (response.status === 404) {
-            throw new Error('Mission not found');
-        }
-        throw new Error('Failed to fetch tags for mission');
+  const response = await fetch(
+    `${FETCH_API_BASE_URL}/missions/tags/${missionId}`,
+    {
+      method: "GET",
+      headers: headers,
+    }
+  );
+  if (!response.ok) {
+    if (response.status === 404) {
+      throw new Error("Mission not found");
     }
-    const data = await response.json();
-    return data.map((tag: any) => ({
-        tagId: tag.id,
-        name: tag.name,
-        color: tag.color,
-    }));
+    throw new Error("Failed to fetch tags for mission");
+  }
+  const data = await response.json();
+  return data.map((tag: any) => ({
+    tagId: tag.id,
+    name: tag.name,
+    color: tag.color,
+  }));
 };
 
 // Get all missions for a tag
-export const getMissionsByTag = async (tagName: string): Promise<{ id: number; name: string; location: string }[]> => {
-    const response = await fetch(`${FETCH_API_BASE_URL}/tags/missions/${encodeURIComponent(encodeURIComponent(tagName))}`, {
-        method: 'GET',
-        headers: headers,
-    });
-    if (!response.ok) {
-        if (response.status === 404) {
-            throw new Error('Tag not found');
-        }
-        throw new Error('Failed to fetch missions for tag');
+export const getMissionsByTag = async (
+  tagName: string
+): Promise<{ id: number; name: string; location: string }[]> => {
+  const response = await fetch(
+    `${FETCH_API_BASE_URL}/tags/missions/${encodeURIComponent(
+      encodeURIComponent(tagName)
+    )}`,
+    {
+      method: "GET",
+      headers: headers,
     }
-    return response.json();
-};
\ No newline at end of file
+  );
+  if (!response.ok) {
+    if (response.status === 404) {
+      throw new Error("Tag not found");
+    }
+    throw new Error("Failed to fetch missions for tag");
+  }
+  return response.json();
+};
+
+// Login system
+export const attemptLogin = async (
+  username: string,
+  password: string
+): Promise<{ success: boolean; cookies?: string }> => {
+  const response = await fetch(`${FETCH_API_BASE_URL}/auth/login/`, {
+    method: "POST",
+    headers: headers,
+    body: JSON.stringify({ username, password }),
+  });
+
+  if (!response.ok) return { success: false };
+
+  let cookies = "";
+
+  Array.from(response.headers.keys()).forEach((key: string) => {
+    const value = response.headers.get(key);
+    if (key.startsWith("set-cookie")) {
+      cookies = `${value}`;
```

Comment:
```suggestion
): Promise<{ success: boolean; cookies?: string[] }> => {
  const response = await fetch(`${FETCH_API_BASE_URL}/auth/login/`, {
    method: "POST",
    headers: headers,
    body: JSON.stringify({ username, password }),
  });

  if (!response.ok) return { success: false };

  let cookies: string[] = [];

  response.headers.forEach((value: string, key: string) => {
    if (key.startsWith("set-cookie")) {
      cookies.push(value)
```
Collect cookies in an array instead of a string